docs/guides/snowflake_integration.md

```diff
@@ -486,3 +505,128 @@ FN_T_UID2_IDENTITY_MAP(EMAIL_HASH, 'email_hash')
+
+The following functions support UID2 sharing:
+
+- [FN_UID2_ENCRYPT](#fn_uid2_encrypt)
+- [FN_UID2_DECRYPT](#fn_uid2_decrypt)
+
+### FN_UID2_ENCRYPT
+
+This function encrypts a raw UID2 and returns the corresponding UID2 token.
+
+Use the applicable prefix to indicate your role:
+- For advertisers: `ADV.FN_UID2_ENCRYPT`
+- For data providers: `DP.FN_UID2_ENCRYPT`
+
+>NOTE: The raw UID2 value is a Base64-encoded string, 44 characters in length.
+
+#### FN_UID2_ENCRYPT: Syntax
+
+The following is the basic syntax for the encrypt function:
+
+```
+Function: FN_UID2_ENCRYPT(raw_uid VARCHAR(44))
+RETURNS VARCHAR(220)
+```
+
+#### FN_UID2_ENCRYPT: Example
+
+The following example illustrates use of this function. The actual change varies according to the specific database, so the `WHERE` is not specified in the example.
+
+```
+SELECT T.token, LATERAL (FN_UID2_DECRYPT(T.token)) AS raw_uid
+FROM receiver's_table T
+WHERE ...
+```
+
+Result: An additional column named `raw_uid` is added to the query result.
+
+```
++------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| raw_uid                                                          | token                                                                                                                                                                                                                        |
+|------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF | AAMAAAACAAECAwQFBgcICQoL18RtKOdx1LlG2zgxK7bTW/nomJIeMFyEJOAMaer+20FTzLBNCJRajrXzXWzb9H3FCtTXs5Rl8PJhYnxxWmMtQ11VHfjI7XiA71CctX2vvCBzD6VEszdNq/t66B1cvLKVjoq2lijm6yyaRi0ovJPuk2dcb8dm4dkHMWGkkOrXbPPvrpsIvYQHH11sbqVycYJZm5w= |
++------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+### FN_UID2_DECRYPT
+
+This function decrypts a UID2 token and returns the corresponding raw UID2.
+
+Use the applicable prefix to indicate your role:
+- For advertisers: `ADV.FN_UID2_DECRYPT`
+- For data providers: `DP.FN_UID2_DECRYPT`
+
+#### FN_UID2_DECRYPT: Syntax
+
+The following is the basic syntax for the decrypt function:
+
+```
+Function: FN_UID2_DECRYPT(token VARCHAR(220))
```

Comment:
Do we mention these lengths (44 and 120) in other places in the docs?

Could they change in future versions (especially future ad token versions)?

Reply:
yes they can definitely change in the future. Actually, the token length 220 is actually for ad token v3 and for latest v4 it should be 218 according to my previous [research ](https://atlassian.thetradedesk.com/confluence/display/~sunny.wu@thetradedesk.com/UID2-526+Advertising+Token+Length+and+URL-Friendliness+Optimisation#UID2526AdvertisingTokenLengthandURLFriendlinessOptimisation-ReducingLengthbyMovingfromBase64EncodingtoBase64URLEncoding). Tho i think we don't need to be absolutely correct and can leave it as 220. 

Who wrote these codes that specify the lengths? Is it possible that we don't specify it to make it flexible for the future? cc @zaiweidu

Reply:
@mcollins-ttd @sunnywu : we do not mention them generally, + we do caution to allow for future changes. I thought it was needed here for the code samples though.
Not sure how to modify these samples based on the comments. Please help. Thx!

Reply:
AFAIK snowflake doesn't document functions this way, it does something like this
https://docs.snowflake.com/en/sql-reference/functions/floor

Define the function
FN_UID2_ENCRYPT(<raw_uid>)

Arguments

<raw_uid> - varchar value of the raw UID value

Returns

Returns encrypted UID2 Tokens in varchar type.



But if we just wanna be quick, i agree with @mcollins-ttd suggestion can follow the same convention as the FN_T_UID2_IDENTITY_MAP definition table:

|Argument|Data Type|Description|
| :--- | :--- | :--- |
| `INPUT` | varchar | The raw UID2 |

A successful query returns encrypted UID2 Tokens in TEXT type. (TEXT is Synonymous with VARCHAR in snowflake see [)](https://docs.snowflake.com/en/sql-reference/data-types-text)

And don't mention the size of varchar